app/views/unattended/provisioning_templates/cloud_init/cloud_init_default.erb

```diff
@@ -42,6 +42,11 @@ users: {}
 runcmd:
 - |
 <%= indent(2) { snippet 'fix_hosts' } -%>
+<% if @host.provision_method == 'image' && root_pass.present? -%>
```

Comment:
I think that you can remove that part: `if @host.provision_method == 'image'` as the template is used for image-based provisioning by default (see line 8 of this file) 

And can you clarify where `root_pass` is coming from?

Reply:
It can come from [hostgroup](https://github.com/theforeman/foreman/blob/develop/app/models/hostgroup.rb#L189-L194) or [host or global settings](https://github.com/theforeman/foreman/blob/develop/app/models/host/managed.rb#L638-L649)

And about the if condition, In my opinion, that should be a separate PR since, the other Userdata and Finish templates also we used that condition for safety purposes example [here](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/finish/kickstart_default_finish.erb#L37-L40) and [here](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/user_data/kickstart_default_user_data.erb#L29-L32)

Those are also used for Image-based deployments only. If we intend to remove the condition for one, the same should be done for others and i would prefer to have a separate PR for the same. But yeah, let me know what you think.

Reply:
> It can come from [hostgroup](https://github.com/theforeman/foreman/blob/develop/app/models/hostgroup.rb#L189-L194) or [host or global settings](https://github.com/theforeman/foreman/blob/develop/app/models/host/managed.rb#L638-L649)

Thanks for the explanation.

> And about the if condition, In my opinion, that should be a separate PR since, the other Userdata and Finish templates also we used that condition for safety purposes example [here](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/finish/kickstart_default_finish.erb#L37-L40) and [here](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/user_data/kickstart_default_user_data.erb#L29-L32)
> 
> Those are also used for Image-based deployments only. If we intend to remove the condition for one, the same should be done for others and i would prefer to have a separate PR for the same. But yeah, let me know what you think.

Here is the PR where the condition was added: https://github.com/theforeman/foreman/commit/c47be8a81e162b5e93ce3e188be9edefb0aa0c96

If I see correctly, this is from a time where there was no separate template for image-based deployment. In my opinion it is safe not to have the check. I agree that if the checks were to be removed this could be done in a follow-up PR.

Reply:
I am marking this resolved for now.